pkg/manifests/rte/rte.go

```diff
@@ -31,62 +32,93 @@ import (
 	"github.com/k8stopologyawareschedwg/deployer/pkg/tlog"
 )
 
-const (
-	NamespaceOpenShift      = "openshift-monitoring"
-	ServiceAccountOpenShift = "node-exporter"
-)
-
 type Manifests struct {
 	ServiceAccount *corev1.ServiceAccount
 	Role           *rbacv1.Role
 	RoleBinding    *rbacv1.RoleBinding
 	ConfigMap      *corev1.ConfigMap
 	DaemonSet      *appsv1.DaemonSet
+
+	// OpenShift related components
+	MachineConfig             *machineconfigv1.MachineConfig
+	SecurityContextConstraint *securityv1.SecurityContextConstraints
+
 	// internal fields
-	plat           platform.Platform
-	serviceAccount string
+	plat platform.Platform
 }
 
-func (mf Manifests) Clone() Manifests {
-	ret := Manifests{
-		plat:           mf.plat,
-		serviceAccount: mf.serviceAccount,
+func (mf *Manifests) Clone() *Manifests {
+	ret := &Manifests{
+		plat: mf.plat,
 		// objects
-		Role:        mf.Role.DeepCopy(),
-		RoleBinding: mf.RoleBinding.DeepCopy(),
-		DaemonSet:   mf.DaemonSet.DeepCopy(),
+		Role:           mf.Role.DeepCopy(),
+		RoleBinding:    mf.RoleBinding.DeepCopy(),
+		DaemonSet:      mf.DaemonSet.DeepCopy(),
+		ServiceAccount: mf.ServiceAccount.DeepCopy(),
+		ConfigMap:      mf.ConfigMap,
 	}
-	if mf.plat == platform.Kubernetes {
-		ret.ServiceAccount = mf.ServiceAccount.DeepCopy()
+
+	if mf.plat == platform.OpenShift {
+		ret.MachineConfig = mf.MachineConfig.DeepCopy()
+		ret.SecurityContextConstraint = mf.SecurityContextConstraint.DeepCopy()
 	}
+
 	return ret
 }
 
 type UpdateOptions struct {
-	ConfigData       string
+	// DaemonSet options
 	PullIfNotPresent bool
-	Namespace        string
+	NodeSelector     *metav1.LabelSelector
+
+	// MachineConfig options
+	MachineConfigPoolSelector *metav1.LabelSelector
+
+	// Config Map options
+	ConfigData string
+
+	// General options
+	Namespace string
+	Name      string
 }
 
-func (mf Manifests) Update(options UpdateOptions) Manifests {
+func (mf *Manifests) Update(options *UpdateOptions) *Manifests {
```

Comment:
we intentionally returned a full copy (not a pointer) to emphasize the function was not modifying the original data

Reply:
Like I can see that all k8s DeepCopy methods generated in the form of
```
func (in *Policy) DeepCopy() *Policy {
	if in == nil {
		return nil
	}
	out := new(Policy)
	in.DeepCopyInto(out)
	return out
}
```

Reply:
sure, that part is fine. My point is more about the methods of the `Manifests` object.
When you have a pointer receiver, you get the message "this method may mutate the receiver object". This is not the case by design about the Manifests object. if we move to pointer receiver we lose this information, every method is potentially mutating even if actually none is (or should). And I'm not sure at all the gains we get are worthy.